pkg/controllers/provisioningv2/rke2/bootstrap/controller.go

```diff
@@ -223,11 +223,14 @@ func (h *handler) getEnvVar(bootstrap *rkev1.RKEBootstrap, capiCluster *capi.Clu
 	return result, nil
 }
 
+// assignBootStrapSecret is utilized by the bootstrap controller's GeneratingHandler method to designate the lifecycle
+// of both the bootstrap secret and related service account. The bootstrap secret and service account must be valid
+// until the corresponding Capi Machine object's Machine Phase is at least "Running", which indicates that the machine
+// "has become a Kubernetes Node in a Ready state".
 func (h *handler) assignBootStrapSecret(machine *capi.Machine, bootstrap *rkev1.RKEBootstrap, capiCluster *capi.Cluster) (*corev1.Secret, []runtime.Object, error) {
-	if capi.MachinePhase(machine.Status.Phase) != capi.MachinePhasePending &&
-		capi.MachinePhase(machine.Status.Phase) != capi.MachinePhaseDeleting &&
-		capi.MachinePhase(machine.Status.Phase) != capi.MachinePhaseFailed &&
-		capi.MachinePhase(machine.Status.Phase) != capi.MachinePhaseProvisioning {
+	if capi.MachinePhase(machine.Status.Phase) == capi.MachinePhaseDeleting &&
+		capi.MachinePhase(machine.Status.Phase) == capi.MachinePhaseDeleted &&
+		capi.MachinePhase(machine.Status.Phase) == capi.MachinePhaseFailed {
```

Comment:
```suggestion
	if capi.MachinePhase(machine.Status.Phase) == capi.MachinePhaseDeleting ||
		capi.MachinePhase(machine.Status.Phase) == capi.MachinePhaseDeleted ||
		capi.MachinePhase(machine.Status.Phase) == capi.MachinePhaseFailed {
```

Reply:
You ever go through the `git commit --amend && git push -u origin --force` and realize you forgot to `git add`? Feel like that happens to me at least once a week. Anyways, should be correct now.